pkg/app/appserver/proc.go

```diff
@@ -259,6 +259,9 @@ func (p *Proc) Stop() error {
 		}
 	}
 
+	// deregister discovery service
+	p.disc.Stop()
```

Comment:
Can `disc` be zero for apps without an updater such as `skychat`?

Reply:
For app like skychat that doesn't need service discovery, we associate [empty updater](https://github.com/skycoin/skywire/blob/2a76c6a19326ff8f78d791d740252895ac15be6a/pkg/app/updatedisc/factory.go#L88) and `Stop()` for those is [empty function](https://github.com/skycoin/skywire/blob/2a76c6a19326ff8f78d791d740252895ac15be6a/pkg/app/updatedisc/discovery_updater.go#L29) too. So derigstration never run for them.